qbft/instance.go

```diff
@@ -56,16 +57,26 @@ func (i *Instance) ForceStop() {
 }
 
 // Start is an interface implementation
-func (i *Instance) Start(value []byte, height Height) {
+func (i *Instance) Start(cdFetcher *types.DataFetcher, height Height, valueCheckF ProposedValueCheckF) {
 	i.startOnce.Do(func() {
-		i.StartValue = value
+		i.CdFetcher = cdFetcher
 		i.State.Round = FirstRound
 		i.State.Height = height
 
 		i.config.GetTimer().TimeoutForRound(FirstRound)
 
 		// propose if this node is the proposer
 		if proposer(i.State, i.GetConfig(), FirstRound) == i.State.CommitteeMember.OperatorID {
+			value, err := cdFetcher.GetConsensusData()
+			if err != nil {
+				fmt.Printf("%s\n", err.Error())
+				return
```

Comment:
The error should be returned and not printed, no?

Reply:
apparently we printed errors in main before...
not sure why...
Maybe we should add errors and test them?
I will open an issue

Reply:
#460